### PR TITLE
Coverage for 'gcloud._apitools.buffered_stream'

### PR DESCRIPTION
**Pull request by tseaver**
~~Uses #1191 as a base.~~

### CODE REVIEW

**Comment by tseaver** [comment]
@dhermes rebased after merge of #1191.

File: gcloud/_apitools/buffered_stream.py [hunk 1]
```diff
@@ -13,32 +13,32 @@ class BufferedStream(object):
...
     def __init__(self, stream, start, size):
-        self.__stream = stream
-        self.__start_pos = start
-        self.__buffer_pos = 0
-        self.__buffered_data = self.__stream.read(size)
-        self.__stream_at_end = len(self.__buffered_data) < size
-        self.__end_pos = self.__start_pos + len(self.__buffered_data)
-
-    def __str__(self):
+        self._stream = stream
```
Comment (dhermes) [inline gcloud/_apitools/buffered_stream.py]: :+1: On this refactor.

File: gcloud/_apitools/buffered_stream.py [hunk 2]
```diff
@@ -13,32 +13,32 @@ class BufferedStream(object):
...
+        self._start_pos = start
+        self._buffer_pos = 0
+        self._buffered_data = self._stream.read(size)
+        self._stream_at_end = len(self._buffered_data) < size
+        self._end_pos = self._start_pos + len(self._buffered_data)
+
+    def __str__(self):  # pragma: NO COVER
```
Comment (dhermes) [inline gcloud/_apitools/buffered_stream.py]: Fine with the `NO COVER` but this should be easy enough to test.

File: gcloud/_apitools/buffered_stream.py [hunk 3]
```diff
@@ -13,32 +13,32 @@ class BufferedStream(object):
...
     def stream_end_position(self):
-        return self.__end_pos
+        return self._end_pos
 
     @property
     def _bytes_remaining(self):
-        return len(self.__buffered_data) - self.__buffer_pos
+        return len(self._buffered_data) - self._buffer_pos
 
     def read(self, size=None):  # pylint: disable=invalid-name
```
Comment (dhermes) [inline gcloud/_apitools/buffered_stream.py]: Is this `disable` actually necessary?

File: gcloud/_apitools/test_buffered_stream.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        CONTENT = b'CONTENT GOES HERE'
+        START = len(CONTENT)
+        BUFSIZE = 10
+        stream = BytesIO(CONTENT)
+        stream.read(START)  # already consumed
+        bufstream = self._makeOne(stream, START, BUFSIZE)
+        self.assertTrue(bufstream.stream_exhausted)
+        self.assertEqual(bufstream.stream_end_position, len(CONTENT))
+        self.assertEqual(bufstream._bytes_remaining, 0)
+        self.assertEqual(bufstream.read(10), b'')
```
Comment (dhermes) [inline gcloud/_apitools/test_buffered_stream.py]: Getters get tested transitively?

**Comment by tseaver** [comment]
Note that I killed off the private attrs/methods in order to ease testing.

File: gcloud/_apitools/buffered_stream.py (hunk 1 above)
Comment (tseaver) [inline gcloud/_apitools/buffered_stream.py]: I would've left it alone, but the private variables made writing tests too hard.

File: gcloud/_apitools/buffered_stream.py (hunk 3 above)
Comment (tseaver) [inline gcloud/_apitools/buffered_stream.py]: Dunno -- it is a carryover from the old codebase.  We can review when we enable `pylint` post-cleanups.

File: gcloud/_apitools/test_buffered_stream.py (hunk 4 above)
Comment (tseaver) [inline gcloud/_apitools/test_buffered_stream.py]: Yup.

Comment (dhermes) [inline gcloud/_apitools/test_buffered_stream.py]: Can we add simple explicit tests for them?

File: gcloud/_apitools/buffered_stream.py (hunk 3 above)
Comment (dhermes) [inline gcloud/_apitools/buffered_stream.py]: OK

File: gcloud/_apitools/test_buffered_stream.py (hunk 4 above)
Comment (tseaver) [inline gcloud/_apitools/test_buffered_stream.py]: They are tested via assertions in the tests for the methods which set the state:   e.g., the `ctor` tests, the `test_read_exhausted`, etc.

File: gcloud/_apitools/buffered_stream.py (hunk 2 above)
Comment (tseaver) [inline gcloud/_apitools/buffered_stream.py]: I'd just as soon rename it `__repr__`, which would be skipped in our `.coveragerc`:  I don't see any value to the actual representation beyond debugging.

File: gcloud/_apitools/test_buffered_stream.py (hunk 4 above)
Comment (dhermes) [inline gcloud/_apitools/test_buffered_stream.py]: Yes I realize. They aren't explicit though.

File: gcloud/_apitools/buffered_stream.py (hunk 2 above)
Comment (dhermes) [inline gcloud/_apitools/buffered_stream.py]: SGTM

Comment (tseaver) [inline gcloud/_apitools/buffered_stream.py]: 56ce426

File: gcloud/_apitools/test_buffered_stream.py (hunk 4 above)
Comment (tseaver) [inline gcloud/_apitools/test_buffered_stream.py]: Don't see the point for the unconditional one-liners.  d7b4ffb splits out the tests for `_bytes_remaining`.

**Comment by dhermes** [comment]
LGTM pending Travis.

**Pull request merged**